[T16-4] Module Planner

rongjiecomputer opened the pull request:
/cc @GabrielYik @Hilda-Ang @RomaRomama

rongjiecomputer committed:
README.adoc: update Travis and Appveyor badges (#2)

GabrielYik committed:
Update README.adoc to describe our product (#3)

* README.adoc: Update to describe our product

* AboutUs.adoc: Update personal profile

* README.adoc: Fix mistakes

* README.adoc: Edit details

RomaRomama committed:
Developer Guide and Profile updated (#5)

* Profile update

* DG update

Hilda-Ang committed:
Update Profile and User Guide to describe current project (#6)

* Add profile image

* AboutUs.adoc: Update profile

* UserGuide.adoc: Update User Guide for current project

* UserGuide.adoc: Update Introduction

rongjiecomputer committed:
Update docs and developer profile (#7)

- Change title of .adoc files to "Module Planner".
- Update URLs to point to this repo.
- Add Rong Jie's profile to AboutUs.adoc.
- Update AsciiDoc variables in build.gradle.

GabrielYik committed:
Update UI mockup in README.adoc (#16)

The new UI mockup image is in docs/images/Ui.png

GabrielYik committed:
Task/to do (#27)

* Model: Add enumerations

Enumerations include ModuleType, FocusArea and Major.

* Add ModuleInfoStorage and related classes

The ModuleInfo class is a empty class created just to stop IDE warnings. The ModuleInfoStorageTest class is created in preparation for future unit tests.

* Fix checkstyle error in ModuleInfoStorageTest

* Format to standardise

* DeleteModuleCommand.java: Update MESSAGE_USAGE

* CliSyntax.java: Add new prefixes

Prefixes include one for the module code, one for the year of study and one for the semester of study.

* ModuleInfoStorage.java: change how module info are stored

List<ModuleType> --> Map<ModuleType, List<ModuleInfo>>

* ModuleInfo.java: format to standardise

* Add ModuleUtil as a container for module helper functions

* Format to standardise

sushinoya commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,42 @@
+
 We are a team based in the http://www.comp.nus.edu.sg[School of Computing, National University of Singapore].
 
 == Project Team
```
Might wish to add responsibilities for each developer as well

sushinoya reviewed (commented):
It is good that you all have updated your User Guide and Developer Guide and seem to have a clear, common idea of what the app should and should not do. Also, basic constructs such as the enums for modules are great. Should start working more on the Java code soon :)

sushinoya commented:
Also, you might wish to start refactoring away from AddressBook to ModulePlanner soon because the longer you all take to get that started, the more you all will drag adding new features.

GabrielYik committed:
Apply upstream bug fix for PersonListPanelTest.java (#29)

See nus-cs2103-AY1819S1/forum#53

src/test/java/seedu/address/ui/PersonListPanelTest.java

GabrielYik committed:
Add dummy delete module command (#28)

* Add DeleteModuleParser.java

Class is complete but not backed by complete implementation

* Modify files to cater to delete module command

* Update instances of parseCode() to parseModuleCode()

* Shift ModuleUtil.java from commons to model

* ParserUtil.java: Adjust import statement ordering

RomaRomama committed:
Created temporary AddModuleCommands and its parser (in-progress) (#32)

* Update .jpg to .png

* Create temporary UserProfile Class

* Fix checkstyles

* Create temporary AddModuleCommand

* Add corresponding parser

* Update AddressBookParser

* Fix checkstyles

* convert .jpg to .png

* Conforming styles

* Delete unnecessary class: UserProfile

GabrielYik committed:
Feature/module manager (#30)

* Add ModuleManager.java

* Add dummy tests

* Fix CI errors

* Shift ModuleUtil from commons to model

* Fix incompatible method names and remove unused import statements

* Delete Module.java

Delete to avoid conflict with the one that Hilda is adding

* Edit import statement for ModuleManager.java

rongjiecomputer committed:
Docs: Rename profile pic so that lecturer's script can pick it up (#35)

rongjiecomputer committed:
Revert "Feature/module manager (#30)" (#36)

This reverts commit e545d76ca19a3c629926445408e8a45270d37398.

Hilda-Ang committed:
Create ModulePlanner and Semester classes (#31)

* Add a placeholder for Module class

* Create Semester class

* Semester.java: Add attributes and constructor

* Semester.java: Create placeholder for methods to be implemented

* Create ModulePlanner class

* ModulePlanner.java: Create placeholder for methods to be implemented

* Module.java & Semester.java: Fix style violations

* Semester.java: Fix indentation

* Convert file format to Unix

* Modify documentation to specify author and add more details

* ModulePlanner.java: Remove trailing whitespace

* Change @author annotation to follow guide and abstract MAX_NUM_SEMESTERS

* ModulePlanner.java: Fix type in documentation

* ModulePlanner.java: Change constant name to fit styleguide

* ModulePlanner.java: Change constant name to fit styleguide

* ModulePlanner.java: Change constant name to fit styleguide

Hilda-Ang committed:
Modify parser for list command (#33)

* Add a placeholder for ListModuleCommand class

* Create and implement ListModuleCommandParser

* ListModuleCommand.java: Add constructor and attributes needed for parsing

* ParserUtil.java: Add parser support for year and semester

* Fix type mismatch for year and semester

* Include listModule command in main parser

* Fix import statements order

* Change file format to Unix

* Change @author annotation to follow guide

* ParserUtil.java: Remove trailing whitespace

GabrielYik committed:
Add ModuleManager (#37)

* Add ModuleManager

* Fix CI errors

GabrielYik committed:
Remove "email us" in Contact Us doc (#43)

rongjiecomputer committed:
Refactor: rename seedu.address to seedu.planner (#41)

GabrielYik committed:
Update "Setting up" section of developer guide (#42)

* Update "Setting up" section of developer guide

* Change planner to address

GabrielYik committed:
Remove unnecessary parts in developer guide (#55)

GabrielYik committed:
Update/house keeping (#56)

* Edit javadocs to fit to team standards and move some classes to different folders

* Add ui mock-up diagram to diagram folder

* Add and edit dummy tests

GabrielYik committed:
Add in some UI backend

* Partially implement the Module and ModuleInformation classes

* ModuleType.java: Add toString() functionality

* ModuleInformation.java: Add Collate author annotation

* Add partially implemented classes and fxmls to view modules in the UI

* Update Logic.java and LogicManager.java

Add in methods needed for use in MainWindow.java

* Update Model.java and ModelManager.java

Add in methods needed for use in MainWindow.java

* Update MainWindow.java to take in dummy modules

* MainWindow.fxml: Undo change

* Remove ModulesTakenListPanel.java first for consistency

* LogicManager.java: Fix compile errors

* MainWindow.java: Remove trailing white space

* Fix checkstyle errors and warnings

* AddCommandTest.java: Add dummy methods to fix compile error

The new methods added in the Model interface were not implemented in the above class.
This commit implements them in a dummy form.

* AddCommandTest.java: Remove trailing white spaces

* MainWindow.java: Uncomment to prevent null error

* AddModuleCommand.java: Bring ending curly bracket down

* Add in javadoc comments and Collate author tags

rongjiecomputer committed:
Rename app title and set app version (#58)

GabrielYik committed:
Add tabs to UI (#60)

* Add ModuleGenerator class

This class holds some sample modules that can be accessed through its methods.

* Remove duplicate ModuleType.java

* ModuleInformation.java: Change private constructor to public

* Add ModuleListCard.java and ModuleListPanel.java

* ModelManager.java: Initialise with sample modules instead of null

* Duplicate and edit MainWindow.fxml

Add an extra column for the available modules

* ModuleGenerator.java: Remove null initialisation

* Link MainWindow.java and MainWindow_Test.fxml

* Change fx ids

* Add in preclusion and prerequisite labels

* ModuleListCard.fxml: Fix truncated preclusion and prerequisite labels

* MainWindow_Test.fxml: Temp fix to make buttons fill screen

* Fix checkstyle errors

* Add lightweight constructors for convenience

Useful for when one needs a module but not with all the attributes initialised

* Replace ModuleGenerator.java with SampleModulesUtil.java

* ModulePlanner.java: Add in new constants

* UI: Add tabs

* Change ModuleInformation to ModuleInfo

rongjiecomputer committed:
Add basic framework to provide ModuleInfo (#59)

- ModuleInformationStorage is renamed to ModuleInfoStorage and convert it to interface so that we can switch between JSON and XML if needed.
- JsonModuleInfoStorage implements ModuleInfoStorage.
- Add moduleConverter.py to download NUSMods JSON file and convert it for Module Planner app (still WIP).
- preclusions and prerequisites of ModuleInfo are now ModuleInfo[] (Module depends on ModuleInfo, so we must make sure ModuleInfo does not depend on Module, otherwise cycle).

sushinoya commented:
<img width="745" alt="screenshot 2018-10-11 at 4 49 42 am" src="https://user-images.githubusercontent.com/23443586/46765252-209b8000-cd11-11e8-918f-6071652bd143.png">

Where is the CLI input bar? Take not that the app should be completely usable through the CLI.

sushinoya commented:
<img width="177" alt="screenshot 2018-10-11 at 4 51 48 am" src="https://user-images.githubusercontent.com/23443586/46765361-66584880-cd11-11e8-9f0e-39f4e69d1532.png">

Could be more specific about roles and responsibilities. Cause as of now, everyone is a `Developer`

sushinoya commented:
Great job getting done with refactoring and implementing some of the commands. Some minor coding standard violations here and there. But good job! 👍

Hilda-Ang committed:
Add suggest command description to User Guide (#61)

* Add suggest command description to User Guide

* Fix suggest command usage example in User Guide

* Remove Data Encryption part from User Guide

GabrielYik committed:
AboutUs.adoc: Add responsibilities (#64)

Hilda-Ang committed:
UserGuide.adoc: Change 'user' to 'you' to address reader (#67)

GabrielYik committed:
UserGuide.adoc: Update delete command syntax (#69)

Hilda-Ang committed:
Add suggest command (#78)

* Create suggestModuleCommand class

* AddressBookParser.java: Add suggestModule command

* SuggestModuleCommand.java: Change file format to Unix

Hilda-Ang committed:
Create IndexUtil class to convert year and semester to an index (#76)

* Parse year and semester inputs to integer

* Create IndexUtil.java to handle year and semester indexing

* ParserUtil.java: Validate year and semester input

* Convert year and semester to index before executing list command

* ParserUtil.java: Fix return type mismatch in parseYear and parseSemester

* IndexUtil.java: Change file format to Unix

GabrielYik committed:
UI: Enhance backend (#65)

* MainWindow_Test.fxml: Remove white space under status bar

* Override toString method in enumerations

The enumerations are Major and FocusArea

* SampleModulesUtil.java: Add in more sample modules

* Change method signatures and corrected implemented methods

Change method signatures to accept a semester index as a parameter

* Small format changes

* Module.java: Add TODO comment

* Semester.java: Add new documentation

Change some documentation as well

* Duplicate some AddressBook classes and modify them to fit in ModulePlanner

* SampleModulePlannerUtil.java: Add to generate sample ModulePlanners

* MainWindow.java: Change semesters to have their own unique modules

* ModuleListCard: Remove unnecessary fx:id

* Remove white space at tab bar and add tab select highlight

* DarkTheme.css: Change blue focus indicator to green

* Change document of Semester

Remove mention of nominal and offset index

* Rename method names from Model.java

* ModelManager.java: Change retrieval method of modules

Modules are now retrieved from the local VersionedModulePlanner instead of a local module list cache

* ModelManager.java: Remove unused imports

* AddCommandTest.java: Remove not overridden methods

Hilda-Ang committed:
Implement list command (#68)

* Implement list command call chain

* ModulePlanner.java: Implement constructor and listModulesTakenForSemester

* ModelManager.java: Specify type argument for FilteredList

* LogicManager.java: Comment out getFilteredTakenModuleList()

* LogicManager.java: Create placeholder for getFilteredTakenModuleList()

* ModelManager.java: Fix incompatible type in getFilteredTakenModuleList method

* LogicManager.java: Fix error in placeholder for getFilteredTakenModuleList()

* ModelManager.java: Fix constructor in getFilteredTakenModuleList method

* LogicManager.java: Add import statement for FXCollections

* ModelManager.java: Initialize all variables in constructors

* ModelManager.java: Initialize all variables in constructors

* AddCommandTest.java: Add getFilteredTakenModuleList method parameters to follow Model

* ModelManager.java: Add line break in constructor to follow style guide

* ModelManager.java: Remove trailing whitespace in constructor

* Implement list command call chain

* ModulePlanner.java: Implement constructor and listModulesTakenForSemester

* ModelManager.java: Specify type argument for FilteredList

* LogicManager.java: Comment out getFilteredTakenModuleList()

* LogicManager.java: Create placeholder for getFilteredTakenModuleList()

* ModelManager.java: Fix incompatible type in getFilteredTakenModuleList method

* LogicManager.java: Fix error in placeholder for getFilteredTakenModuleList()

* ModelManager.java: Fix constructor in getFilteredTakenModuleList method

* LogicManager.java: Add import statement for FXCollections

* ModelManager.java: Initialize all variables in constructors

* ModelManager.java: Initialize all variables in constructors

* AddCommandTest.java: Add getFilteredTakenModuleList method parameters to follow Model

* ModelManager.java: Add line break in constructor to follow style guide

* ModelManager.java: Remove trailing whitespace in constructor

* Change year and semester to index

* ModulePlanner.java: Change year and semester usage in listModulesTakenForSemester method to index

* AddCommandTest.java: Change year and semester to index

* Fix method name mismatch and add constructor placeholder

* ModelManager.java: Add constructor placeholder and uncomment methods

* ModelManager.java: Initialise versionedAddressBook and filteredPersons in constructor

* LogicManager.java: Remove unused import statement

* Change methods' names to be more concise

* ReadOnlyModulePlanner.java: Fix method name mismatch

* ReadOnlyModulePlanner.java: Fix javadoc param names mismatch

rongjiecomputer committed:
Expose Model.getModuleInfo() (#80)

- Expose ModuleInfo[] Model.getModuleInfo() so that other commands can use it.
  (Might expose code->ModuleInfo map in the future too).
- Module credit is now float (because we have modules with *.5 MC).
- Add processed data/moduleInfo.json with fake data.
- ModuleInfo's second constructor now receives arrays of strings for prereq and preclu instead
  of arrays of ModuleInfos due to certain design limitation. This constructor should only be
  used in test. In most cases, you should get ModuleInfo from Model.getModuleInfo() instead.

```
    public ModuleInfo(String code, String name, ModuleType[] possibleTypes,
                      float creditCount, String[] preclusions, String[] prerequisites) {}
```

Other notable changes:

- Update Google Guava to 21.0 (for ImmutableList.toImmutableList).

Hilda-Ang committed:
README.adoc: Update coveralls badge (#83)

Hilda-Ang committed:
Implement parser for suggest command (#81)

* Create SuggestModuleCommandParser

* SuggestModuleCommandParser.java: Change file format to Unix

* Use parser support to handle suggest command

* ModulePlanner.java: Fix index in listModulesAvailable

* Make attribute index private

GabrielYik committed:
Update User Guide (#84)

* UserGuide.adoc: Update outdated delete command example

* UserGuide.adoc: Update ui picture

GabrielYik committed:
Revert "README.adoc: Update coveralls badge" (#87)

* Revert "Update User Guide (#84)"

This reverts commit 7787873634cf60c7a3304c2f6d08d62d0dbb21ef.

* Revert "Implement parser for suggest command (#81)"

This reverts commit fd9a5aed2d90466b35671611b03244b7d35dc2cf.

* Revert "README.adoc: Update coveralls badge (#83)"

This reverts commit fabff9ba4129388f72cad59b59a54d4159941014.

GabrielYik committed:
UserGuider.adoc: Fix display failure of ui.png (#89)

* UserGuide.adoc: Update outdated delete command example

* UserGuide.adoc: Update ui picture

* README.adoc: Fix display failure of ui.png

Hilda-Ang committed:
Add description of the implementation of list feature in Developer Guide (#82)

* DeveloperGuide.adoc: Add list feature implementation details

* DeveloperGuide.adoc: Remove unnecessary whitespace

GabrielYik committed:
Add description for delete feature implementation (#79)

* Docs: Remove unnecessary pictures

* DeveloperGuide: Add delete feature implementation

In addition, change all mentions of address book to module planner

* DeveloperGuide: Fix typos

* DeveloperGuide.adoc: Update method name in Delete Feature portion

rongjiecomputer committed:
DeveloperGuide.adoc: Update doc for Storage (#86)

* DeveloperGuide.adoc: Update doc for Storage

* Update .pptx

* Document ModuleInfo

GabrielYik committed:
UserGuide.adoc: Potential fix for ui.png display issue (#91)

* UserGuide.adoc: Update outdated delete command example

* UserGuide.adoc: Update ui picture

* README.adoc: Fix display failure of ui.png

* README.adoc: Potential fix for ui.png display failure

RomaRomama committed:
Add description of add feature implementation in DG (#88)

* Add implementation in DG

* DeveloperGuide.adoc: Update doc for Storage (#86)

* DeveloperGuide.adoc: Update doc for Storage

* Update .pptx

* Document ModuleInfo

* UserGuide.adoc: Potential fix for ui.png display issue (#91)

* UserGuide.adoc: Update outdated delete command example

* UserGuide.adoc: Update ui picture

* README.adoc: Fix display failure of ui.png

* README.adoc: Potential fix for ui.png display failure

GabrielYik committed:
docs: Update UI picture (#103)

GabrielYik committed:
Implement Delete Command (#97)

* Remove duplicate DeleteModuleParser

* Messages.java: Add new message

* StringUtil.java: Add method to check if a string is empty

* ModuleUtil.java: Properly implement module code format checking

The format of the module code is checked against a regex

* Semester.java: Add new methods

Change implementation of some methods

* ModuleInfo.java: Edit MESSAGE_MODULE_CODE_CONSTRAINTS

* Module.java: Override equals() and toString() methods

* ParserUtil.java: Properly implement parseModuleCodes() method

* AddModuleCommand.java: Change constructor signature

* Add new methods and edit existing

These methods pertain to the manipulation of modules and the checking of their existence

* DeleteModuleCommandPaser.java: Implement parser

* DeleteModuleCommand.java: Implement execute() method

* AddCommandTest.java: Override new methods from Model.java

* Fix checkstyle errors

* Model.java: Remove trailing whitespace

* AddCommandTest.java: Remove trailing whitespace

* ModuleUtil.java: Update regex

* Add tests for util methods

* ModuleUtilTest.java: Remove unused import statement

* ModulePlanner.java: Fix bug

Fix the improper setting of semesters and possible out-of-bounds getting of semesters

* Fix to display temporary modules

* Fix checkstyle error

RomaRomama committed:
Implement add command  (#106)

* Ad addMoules to Model and ModelManager

* add addModules to Model and ModelManager

* ParserUtil: Change return type of parseModuleCodes into List<Module>

* DeleteModuleCommandParser: Modify accordingly to the new parser

* Fix some things

* implement addModuleCommandParser

* implement AddModuleCommand

* Modify ModelStub

* Add override

GabrielYik committed:
Implement goto command (#107)

* TabSwitchEvent.java: Add event for tab switching

* Add in bridging classes

* Implement Goto command

* Change base semester from 1 to 0 for convenience in coding

* Remove unused import statements and add documentation

* StringUtil.java: Add in new method

The method checks if a String is made up of only alphabets

* Some formatting

* ParserUtil.java: Remove unused method

* Fix checkstyle errors

* Add documentation

sushinoya commented:
The developer guide still mentions how to delete/add Persons.

<img width="664" alt="screenshot 2018-10-22 at 3 41 41 pm" src="https://user-images.githubusercontent.com/23443586/47282254-192d7e00-d611-11e8-986e-20047fc581ae.png">
<img width="1032" alt="screenshot 2018-10-22 at 3 41 18 pm" src="https://user-images.githubusercontent.com/23443586/47282260-1af74180-d611-11e8-9572-c45905a1f388.png">

sushinoya commented:
The developer guide still mentions how to delete/add Persons.

<img width="664" alt="screenshot 2018-10-22 at 3 41 41 pm" src="https://user-images.githubusercontent.com/23443586/47282254-192d7e00-d611-11e8-986e-20047fc581ae.png">
<img width="1032" alt="screenshot 2018-10-22 at 3 41 18 pm" src="https://user-images.githubusercontent.com/23443586/47282260-1af74180-d611-11e8-9572-c45905a1f388.png">

sushinoya commented:
The developer guide still mentions how to delete/add Persons.

<img width="664" alt="screenshot 2018-10-22 at 3 41 41 pm" src="https://user-images.githubusercontent.com/23443586/47282254-192d7e00-d611-11e8-986e-20047fc581ae.png">
<img width="1032" alt="screenshot 2018-10-22 at 3 41 18 pm" src="https://user-images.githubusercontent.com/23443586/47282260-1af74180-d611-11e8-9572-c45905a1f388.png">

sushinoya commented:
The developer guide still mentions how to delete/add Persons.

<img width="664" alt="screenshot 2018-10-22 at 3 41 41 pm" src="https://user-images.githubusercontent.com/23443586/47282254-192d7e00-d611-11e8-986e-20047fc581ae.png">
<img width="1032" alt="screenshot 2018-10-22 at 3 41 18 pm" src="https://user-images.githubusercontent.com/23443586/47282260-1af74180-d611-11e8-9572-c45905a1f388.png">

rongjiecomputer committed:
MainApp.java: Actually use data from moduleInfo.json (#109)

* MainApp.java: Actually use data from moduleInfo.json

Used to always use empty array even when we successfully found the file

GabrielYik committed:
Implement SetUp Command (#108)

* Model.java: Add new methods

Implement new methods in ModelManager as well

* Add new UNKNOWN enums

* CliSyntax.java: Add prefixes for major and focus area

* ParserUtil.java: Add new util methods

* MainWindow.java: Add switch tab event for goto command

* IndexUtil.java: Rename boolean methods

* AddressBookParser.java: Add SetUp Command case

* GoToCommandParser.java: Add author annotation

* Add classes related to SetUpCommand

Includes SetUpCommand, SetUpCommandParser and UserProfile

* AddressBookParser.java: Remove duplicate commands

* Fix ListView failure to update modules when modules are deleted

* StringUtil.java: Update regex and add new method

* SetUpCommand.java: Update messages

* ParserUtil.java: Update method names used in util methods

* Fix checkstyle errors

* AddCommandTest.java: Override new methods

* SetUpCommand.java: Fix typo

* Override equals()

* StringUtil.java: Update regex to be stricter

* Update a ParserUtil method and its usage

* DeleteModuleCommandParser.java: Update command syntax to use prefix

* IndexUtil.java: Remove convertToZeroBased()

* ModuleInfo.java: Add requireAllNonNull() to constructor

* DeleteModuleCommand.java: Fix indent

Hilda-Ang committed:
Add sequence diagram for list command and change index to zero-based (#110)

GabrielYik committed:
Miscellaneous additions (#111)

* CommandTestUtil.java: Add new constants for convenience

* ParserUtil.java: Add requireNonNull()

* Add isModuleOffered() and finalizeModule()

The finalizeModule() returns the module with the actual module information

* StringUtilTest.java: Add new tests

* AddCommandTest.java: Override new method

* Rename ModuleInformationTest.java to ModuleInfoTest.java

* Update GoToCommand to use prefixes

* Remove Tab class and other small changes

* Migrate sample modules and sample moduleplanner generator to test util

* Add tests

Tests are for the DeleteModuleCommandParser, the GoToCommandParser and the SetUpCommandParser

* Fix modules added not showing up on ui

* Fix checkstyle errors

* Fix checkstyle error

GabrielYik commented:
> Great job getting done with refactoring and implementing some of the commands. Some minor coding standard violations here and there. But good job! 👍

What kind of coding standard violations were there?

rongjiecomputer committed:
Rework ModuleInfo and implement ModulePlannerStorage (#112)

- Move `/data/moduleInfo.json` into `resource` folder and package it in `.jar` file. Since it is now a resource file, we no longer read it from `ModuleInfoStorage`. Instead, we have a singleton class `ModuleInfoRetrierver` to handle this (implementation detail).
- Implement `[Json]ModuleInfoStorage` to store the whole `ModulePlanner` (which has a list of `Semester`).

Note for Command implementer:
- Get complete list of `ModuleInfo` from `Model.getModuleInfo`.
- Use `Optional<ModuleInfo> ModuleInfo.getFromModuleCode(String code)` to transform a module code into `ModuleInfo` (so no need to linear search the whole array).

Drive-by fixes:
- Rename `addressbook.jar` to `moduleplanner.jar`.
- Bump version to `1.3.0`.

Hilda-Ang committed:
Add tests for ListModuleCommandParser (#114)

* ParserUtilTest.java: Add tests for new methods

* ParserUtil.java: Fix typo in parseSemester

* ParserUtil.java: Add requireNonNull in parseMajor

* Create tests for ListModuleCommandParser

* Add tests for ListModuleCommandParser

* CommandTestUtil.java: Fix typo in constant

* Fix type mismatch and change methods access

* ListModuleCommand.java: Add equals method

* Modify parseModuleCode and parseFocusArea to private

* ListModuleCommandParserTest.java: Fix author tag

Co-Authored-By: Hilda-Ang <E0196709@u.nus.edu>

rongjiecomputer committed:
Fix not finalizing ModuleInfo and actually hook ReadOnlyModulePlanner… (#115)

* Fix not finalizing ModuleInfo and actually hook ReadOnlyModulePlanner in ModelManager

rongjiecomputer committed:
Fix ModulePlanner.hasModule (#116)

rongjiecomputer committed:
Fix saveModulePlanner to save to correct file path (#121)

rongjiecomputer committed:
Fix resource loading bug in .jar file and add modulePlannerPath to userPrefs (#122)

Hilda-Ang committed:
Refactor model, commands, and tests (#113)

* Refactor Model and ModelManager from AddressBook to ModulePlanner

* Remove unnecessary commands

* Refactor addressBook to modulePlanner

* Remove unused method

* MainWindow.java: Refactor UI

* Remove unused import statements and fix indentation

* Remove irrelevant tests

* Remove unused imports

* Remove irrelevant tests

* Modify and remove irrelevant tests

* Modify and remove irrelevant tests

* TestApp.java: Fix typo in getModel

* Modify irrelevant tests

* Fix incompatible types

* Fix checkstyle error

* Add equals and hashCode methods

* Comment out irrelevant tests

* Remove unused imports

* Fix failing tests

* Fix failing tests

* Comment out failing tests

* Remove unused imports

* Remove unused imports

* Comment out failing tests

* Fix indentation level

* Create testutils and modify existing tests

* Fix failing test

* Fix array initialization

* Fix bugs

* Add missing semicolon

* Fix bugs

* Fix import statements order

* Fix bugs

* Fix checkstyle violations

* Change file format to Unix

* Remove irrelevant files and comment out irrelevant tests

* Remove unused method call

* Remove unnecessary classes

* Remove deleted class usage

* Comment out failing tests

* Remove unused import statements

* Remove AddressBook and refactor related classes

* Remove unused import statement

* Import Paths

* Comment out failing test

* Add dummy to gitignore

* Delete dummy in main

Hilda-Ang committed:
Add config.json for RepoSense and modify gitignore (#124)

* Add config.json for RepoSense and modify gitignore

* Update author name

Co-Authored-By: Hilda-Ang <E0196709@u.nus.edu>

Hilda-Ang committed:
Remove remaining unused classes and update relevant parts to modulePlanner (#129)

* Remove Person and related classes

* Remove unnecessary tests and modify addressBook to modulePlanner

* Remove unnecessary prefixes

* Remove irrelevant part from Developer Guide

GabrielYik committed:
Update gh-pages (#132)

* Remove unrelated elements

* Update website url to trial changes

* Revert url to team url

* Remove learning outcomes section

GabrielYik committed:
Implement Find Command (#130)

* Remove unnecessary pic

* MainWindow.java: Change file path

The aim of this commit is to show that data is saved to modulePlanner.json instead of addressBook.json in the Status Bar

* Add associated classes for Find command

* Remove the word "module" from class names

Also, update the command word for the suggest command from suggestCommand to suggest

* ModuleInfo.java: Update message constraints

* Rename package containing FocusArea.java and Major.java

Rename to "course" to better reflect the purpose of the classes

* Remove unnecessary sample util classes

* ModelPlanner.java: Set setSemesters() to private

The method was required to be exposed previously but does not need to now

* Move classes associated with the user into the package "user"

* Remove duplicate classes

* ModelManager.java: Change import statement for UserProfile

* ModelPlanner.java: Revert method access from private to public

* ModulePlannerParser.java: Arrange commands in alphabetical order

* Add new events for new commands

FindModuleEvent is for the Find command. StatusEvent is for the Status command

* Remove unnecessary PersonPanelSelectionChangedEvent.java

* Remove the word "filtered" from Logic methods

* Add classes associated with the Find command

* Remove the word "filtered" from Model methods

* Add support for Find command in UI

* ParserUtil.java: Include explicit String trimming for some methods

* DeleteCommand.java: Add a space behind the MESSAGE_USAGE

* MainWindow.java: Add handler for find command event

* Fix checkstyle errors

* SetUpCommand.java: Prevent invalid parameters from being accepted

* CommandTestUtil.java: Fix method name change

rongjiecomputer committed:
Redesign ModuleInfo API (#137)

Get data from static functions (`ModuleInfo.getModuleInfoList` and `ModuleInfo.getFromModuleCode`) without asking from `Model[Manager]`.

Some test helper functions are removed so that I can remove `ModuleInfo(String code, String name, ModuleType[] possibleTypes, float creditCount, String[] preclusions, String[] prerequisites)` constructor that is not safe to use (need to know implementation detail of `ModuleInfo`).

RomaRomama committed:
Add temporary StatusCommand, fix addCommand bug, add addCommandParser tests  (#138)

* Fix bug of being able to add an existing module

* Add AddCommandParserTest

* Add temporary status command

* fix checkstyle

* Fix checkstyle 2

* Remove failed test

* Minor changes

* Fix checkstyle 3

GabrielYik committed:
Add tests for some commands (#136)

* Remove unnecessary ModuleManager.java

* Add new StringUtil methods and corresponding tests

* Messages.java: Update format specifier for consistency

* FindCommand.java: Override equals() method

* SetUpCommand.java: Adjust formatting of MESSAGE_SET_UP_SUCCESS

* Update classes to use newly added StringUtil methods

* Major.java: Capitalise the first letter of each word of each major

* Add tests for some commands

The commands are delete, setup, goto and find

* Fix checkstyle errors

* SetUpCommandParser.java: Make focus area not a compulsory argument

* MainWindow.java: Add closing author annotation

* StringUtilTest.java: Fix broken test code

* Fix checkstyle errors

* Fix checkstyle errors

* Remove unnecessary ModuleManagerTests.java

* GoToCommand.java: Fix goto command not throwing exception

Change the && to ||

* Add more test cases

RomaRomama committed:
Add override method, add some tests and fix variables in AddCommandParserTest (#140)

* Add override method and add some tests to AddCommandParserTest

* Fix Checkstyle

* Fix checkstyle 2

GabrielYik committed:
Update User Guide to reflect current project state (#139)

GabrielYik committed:
Fix inconsistent ordering of focus areas due to use of Set (#141)

* StringUtil.java: Fix inconsistent result of convertCollectionToString()

* SetUpCommandTest.java: Order arguments in alphabetical order

* UserProfile.java: Change mapFocusAreas() return value from a set to a sorted list

* Remove unused imports

* Remove unused import

GabrielYik committed:
Add some modules that fulfil Computer Science requirements (#131)

* Add some modules that fulfil Computer Science requirements

* computerscience.json: Add new line

* Split PC1221/X into two modules

Co-Authored-By: GabrielYik <26066169+GabrielYik@users.noreply.github.com>

* computerscience.json: Fix module codes

* Add in modules needed to fulfil software engineering focus area

Hilda-Ang committed:
Implement suggest command (#125)

* Remove SuggestModuleCommandParser

* Remove paramater index from getFilteredAvailableModuleList and rename some methods

* Remove getModulesAvailable() and hasBeenTaken attribute from Semester class

* Implement methods to suggest modules

* Fix compile errors

* Remove undeclared methods

* Remove unused import statement

* Fix checkstyle violations and add javadocs

* Fix bug unable to display modules

* Add ModuleInfo constructor from module code

* Remove keywords this and nominal

Hilda-Ang committed:
Update coveralls and author name (#144)

* Update coveralls badge

* Add author name for reposense

Hilda-Ang committed:
Add tests for list command (#143)

* Add tests for list command

* Remove unused import statements

* Remove unused import statements

* Update javadoc comments

Hilda-Ang committed:
Add test for suggest command (#146)

* Add test for suggest command

* Fix checkstyle violation

rongjiecomputer committed:
Update reposense authorNames (#150)

GabrielYik committed:
Update developer guide (#145)

* Remove DummySearchPage.html

* Move README.adoc to docs folder

* Update some diagrams

* Rename ModuleListCard to ModuleCard

* DeveloperGuide.adoc: Order and update information

* ModelClassComponentClassDiagram.pptx: Update to reflect project

* ModuleInfo.java: Reorder statements to follow standard

* Revert README.adoc to original place

* Remove trailing whitespace

rongjiecomputer committed:
DeveloperGuide.adoc: Update ModuleInfo and Storage docs (#149)

rongjiecomputer committed:
Improve SetupCommand and implement UserProfile storage (#148)

Main changes:

- `UserProfile` is now an immutable class instead of singleton.
- `UserProfile` is saved in `ModulePlanner` class.
- `Model#setUpUserProfile` will trigger `ModulePlannerChangedEvent`.
- More fine-grained error messages for `setup` and update tests.

Drive-by fixes:
- Move `hasMajor` and `hasFocusAreas` to `Major` and `FocusArea` instead since they can be made static functions and do not rely on the state of `Model`.
- `Model#getModuleInfo` is removed. Use `ModuleInfo.getModuleInfoList` (static function instead).

GabrielYik committed:
Fix undo, redo and clear commands (#151)

* Enable add and delete commands to be committed

* Fix undo and redo commands not working

The problem is that whenever a commit is made to the versioned module planner,
the module planner added into the modulePlannerStateList is not a deep copy. Hence,
even though different commands may be performed, all the module planners stored in
the modulePlannerStateList have the same list of semesters.

The problem was fixed by making sure that the module planner added
to the modulePlannerStateList is a deep copy

* Remove unused imports

* Fix failing delete command test

* ModulePlannerChangedEvent.java: Update toString()

* JsonSerializableModulePlanner.java: Remove trailing white spaces

rongjiecomputer committed:
Update _config/reposense authorNames (#154)

GabrielYik committed:
Fix adding/ deleting of identical modules (#153)

* Remove unnecessary classes and their method usages

* Replace use of List<Module> to Set<Module>

This is to prevent identical modules being specified to be added or deleted

* Add new CollectionUtil methods

The first method added is getAnyOne(), which returns an empty Optional if the collection is empty or an Optional of any element (the first element if the collection is ordered).

The second method added is areEqualIgnoreOrder(), which checks if two collections have identical elements regardless of their order.

* Remove unused imports

* Remove unused imports

* Fix import statement ordering

Hilda-Ang committed:
Change images for Developer Guide (#157)

GabrielYik committed:
Add note about suggest command in UserGuide (#158)

* UserGuide.adoc: Add note on suggest command

* Replace old address book icon with new module planner icon

* Rename some images

Hilda-Ang committed:
Fix available modules list not updated after add and delete command (#155)

* Fix available modules list not updating on add command

* Change file format to Unix

* Remove unused import and remove redundant method

* Fix available modules list not updating on delete command

* Add tests for UniqueModuleList

* Remove unused method

* Remove unused method

* Remove unused import

* Remove keyword this

Hilda-Ang committed:
Add tests for shared classes (#147)

* Add tests for getModulesAvailable()

* Add tests for ModulePlannerParser

* Add tests for LogicManager

* Fix tests for ModulePlannerParser

* Remove unused imports and fix import order

* Add tests for IndexUtil

* Add tests for IndexUtil

* Add tests for ModelManager and fix IndexUtil tests

* Fix import order

* Remove unnecessary tests

* Remove unused import statement

* Add tests for Semester and other classes

* Fix checkstyle violations

* Add tests for ModuleUtil

* Fix import order

* Add tests for Semester

* Fix failing tests

* Fix import order

Hilda-Ang committed:
Remove year and semester arguments from setup command (#161)

* Remove year and semester from user profile

* Update setup command tests

* Add test for setup command

Hilda-Ang committed:
Add parser for suggest command and update based on semester (#160)

* Fix available modules list not updating on add command

* Change file format to Unix

* Remove unused import and remove redundant method

* Fix available modules list not updating on delete command

* Add tests for UniqueModuleList

* Remove unused method

* Remove unused method

* Remove unused import

* Remove keyword this

* Remove UniqueModuleList and related classes

* Add arguments year and semester to suggest command

* Update UI for suggest command event

* Store index for available module list

* Add event handler for go to command

* Clear available modules list on go to command

* Fix checkstyle errors

* Add tests for suggest command and parser

* Change file format to unix

* Fix failing tests

* Remove SuggestModulePanel

* Change currentIndex to index

rongjiecomputer committed:
Fix helpwindow.html not loading (#184)

Revived code that was accidentally deleted in major refactoring #113.

Fixes #176
Fixes #166
Fixes #165
Fixes #164

GabrielYik committed:
Update UI and enhance delete feature (#163)

* ModelManager.java: Update comment

* Rename method name to remove mention of "list"

* Add method to convert a valid index to year and semester

* Remove tabs in ui

* Update how ui responses to some commands

* Fix modules with small case prefixes being disallowed from being added

* StringUtil.java: Add method to check if a string contains only numbers

* Rename some methods for consistency and remove mention of internal data structure

* Use convertCollectionToString method for add and delete commands

* StringUtil.java: Update REGEX name

* Move arePrefixesPresent method into ArgumentMultiMap

Also add a check for prefixes which have more than one values associated when they should have only one

* CliSyntax.java: Add a method to check if a prefix has only one associated value.

This is to prevent prefixes, which are meant to have only one value, from having multiple

* ParserUtil.java: Add a check to make sure that the year and semester are numbers

* Make the modules that appear in the ui sorted

* Update delete feature

Delete now removes modules when their prerequisites are removed. This results in chain deletion of further modules up the semesters

* Update tests

* Remove ModuleInfoTest.java

* Remove unused constructors

* Remove unnecessary predicate

* DarkTheme.css: Add new line

* Remove unnecessary code

* Add javadoc comments and fix checkstyle errors

* ModelManager.java: Fix checkstyle error

* Fix checkstyle errors

* Add additional tests for new methods

* Rename containsPrefixes() to containsAllPrefixes()

rongjiecomputer committed:
Add module description (#185)

Drive-by fix: delete some dead codes and refactor resource JSON file loading to JsonUtil.

RomaRomama committed:
Enhance AddCommand  (#174)

* Move specific class message

* Add getInfo for Module Class

* Add function to detect equivalent modules and moduleinfos

* Enhance AddCommand to display reason why some modules are not added

* Fix checkstyle

* Minor fix mentioning unfulfilled module more than once

* Add AddCommand Tests

* Fix checkstyle

* Fix checktyle

* Created a temporary substitute because test failed with typicalmodules

* Modify so that AddCommand can also take in immutable sets

* Fix unimported variables

* Fix Checkstyle

* Adjust to recent master and suggestion

GabrielYik committed:
Bug fixes (#186)

* Prevent more than one module from being specified for the find command

* Add in description of module in ui

* Add tests for new ArgumentMultimap methods

* Add test for new method in CliSyntax

* DeleteCommand.java: Add ... to MESSAGE_USAGE

This is to show that more than one module can be deleted

* Remove unused imports

* UserGuide.adoc: Update description for setup command

* Fix checkstyle error

* Add new test for FindCommandParser

GabrielYik committed:
Rename ui.png to Ui.png and upload personal PPP (#188)

* Rename ui.png to Ui.png

* Upload PPP

GabrielYik committed:
Enhance delete command (#187)

* Move some StringUtil methods to CollectionUtil

* Enhance delete command

* Remove unused import

* Fix checkstyle errors

Hilda-Ang committed:
Modify list command (#189)

* Add methods to list modules all and by year

* Implement internal list to store modules taken

* Create method in IndexUtil and add list modules methods to ModelManager

* Add list method to LogicManager

* Modify list command arguments

* Create UI for list command

* Fix parser for list command

* Fix checstyle errors

* Modify list command tests

* Change file format to unix

* Add tests

* Remove unused import

* Merge from master and fix style

Hilda-Ang committed:
Fix suggest command prerequisites checking (#191)

* Fix prerequisites checking until current semester

* Add tests for suggestModules method

* Remove trailing whitespace

RomaRomama committed:
Adjust add and delete to behave similarly  (#192)

* Adjust addCommand to have better OOP style and give similar output as deleteCommand

* Modify so DeleteCommand also take immutable set

* Minor fix

* Fix checkstyle

* fIx CHecKstYle

* remove unused import

rongjiecomputer committed:
SuggestCommand: sort suggested modules based on faculty modules and predetermined importance (#190)

After the prereq/preclu filtering, suggested modules will be sorted with stable sort with these criteria:
- Sort modules from a faculty/major (e.g. `CS`, `IS`, `MA` first before anything else).
- Sort modules based on a hard-coded list.

Sorting with `FocusArea` is not yet implemented.

Drive-by-fix:
- Remove dead test files for `ModuleInfoStorage` (no longer exists).
- Some code reformatting.

rongjiecomputer committed:
moduleInfo.json: Ensure backward preclusion relationship (#193)

Drive-by-fix (hack?): Remove prereq for CS1231 and MA1521 to meet most CS students' expectation

rongjiecomputer committed:
moduleInfo.json: Remove all MA1301[X] as prereq (#194)

Hilda-Ang committed:
Fix list not update on add/delete (#195)

* Add methods to list modules all and by year

* Implement internal list to store modules taken

* Create method in IndexUtil and add list modules methods to ModelManager

* Add list method to LogicManager

* Modify list command arguments

* Create UI for list command

* Fix parser for list command

* Fix checstyle errors

* Modify list command tests

* Change file format to unix

* Add tests

* Remove unused import

* Merge from master and fix style

* Update index to availableIndex in setAvailableModules

rongjiecomputer committed:
Fix removing MA1301[X] from preclu instead of prereq (#197)

rongjiecomputer committed:
moduleInfo.json: remove QET modules (#200)

ES1000 and ES1103 are for students who do not pass QET.

Hilda-Ang committed:
Revert tests (#201)

* Revert UI tests

* Revert change to TestApp

* Remove unused import statement

* Revert tests in ClearCommandSystemTest

* Fix checkstyle violations

* Fix failing tests

* Add todos and fix inconsistencies

* Fix checkstyle violations

* Comment out irrelevant test

* Comment out some tests

* Remove debugging statements

rongjiecomputer committed:
Test storage (#202)

Add tests for the following components and fix some bugs discovered by the tests:
- `StorageManager`
- `JsonModulePlannerStorage`
- `JsonSerializableModulePlanner`
- `JsonAdaptedModule`
- `JsonAdaptedSemester`

Resolves #100

GabrielYik committed:
Miscellaneous (#203)

* Add time indicators to "Taken Modules" column for List command

* The content in the Find module panel stretches accordingly

* ModuleListPanel.java: Replace empty String with constant

* LogsCenter.java: Change name of log file

* Remove unused Event and associated code

* ListModuleEvent.java: Remove unused import statement

* MainWindow.java: Add javadoc for initSuggestedModulesPanel

* Use new Pair class for returning values that should come in pairs

* Pair.java: Add javadoc comment

* Migrate some messages from Messages to the classes where they are used

* ModuleDescription.java: Refactor code for setting grammar

If preclusion and prerequisite labels have different grammar structures depending on the number of items they are representing

* Replace MESSAGE_INVALID_MODULES with MESSAGE_NOT_OFFERED_MODULES

* Rename some test methods and add one more

* Remove ModuleBuilder because it does not seem needed

* Remove unused import statements

* TypicalModules.java: Remove unused import statements

* Remove author annotations from interfaces.

Code smaller than a method are not graded

* ModuleListPanel.java: Remove unnecessary code

* ModuleListPanel.java: Remove unused import statements

Hilda-Ang committed:
Update portfolio and User Guide (#204)

* Add project portfolio page

* Rename portfolio and update User Guide

* Update portfolio

rongjiecomputer committed:
Update UG/DG and add first draft of my portfolio (#205)

- Add our portfolio links to `AboutUs.adoc`.
- Add sequence diagram for storage.
- First draft of my portfolio.
- Revive UG for help, clear, undo, redo and history.
- Add field restriction table in UG.
- Update setup's UG.
- Explicitly list down major and focus area supported by setup and suggest.
- Update UI.png.
- Add screenshot for find command.

GabrielYik committed:
Update Developer Guide (#208)

* Rename event class

* DeleteSequenceDiagram.pptx: Adjust arrow positions

* Add sequence diagrams for goto and find commands

* Rename usages of events

* DeveloperGuide.adoc: Update existing and add new descriptions

Update descriptions for delete command and add new descriptions for find and goto commands

* Add new sequence diagram imgs and update delete sequence diagram img

* DeveloperGuide.adoc: Add new images

Hilda-Ang committed:
Add java docs and rename some methods (#206)

* Rename methods and add java docs

* Fix style violations

* Update semesters to years

* Remove java docs for overriden methods

* Remove attached diagram

* Rename modulesTakenUntilIndex to modulesTakenBeforeIndex

Hilda-Ang committed:
Update PPP and UG/DG (#210)

* Add project portfolio page

* Rename portfolio and update User Guide

* Update portfolio

* Update Developer Guide

* Update portfolio and User Guide

* Remove duplicated ListSequenceDiagram_1

GabrielYik committed:
Update User Guide (#209)

* UserGuide.adoc: Update existing information

* Add in structure explanation of ui

* Add img for UiLabelled diagram

* UserGuide.adoc: Complete description for time period

* Add in images for illustration

RomaRomama committed:
StatusCommand done, Tests In Progress (#199)

Improve Status Command

GabrielYik committed:
Miscellaneous (#211)

* ModulePlannerChangedEvent.java: Update toString() message

* Add logging messages to personal commands

* MainWindow.java: Add logging messages to event handlers

* GoToCommandParser.java: Remove code that was commented out

* CollectionUtil.java: Use method references

* UserGuide.adoc: Remove " ` " from non-code and command output segments

* Make the listed modules sorted

* MainWindow.java: Extract out listed modules panel initialisation logic

* Remove code associated with card selection

* MainWindow.java: Rename method name

* DeleteCommand.java: Remove unnecessary logic

There is no need to check if the modules to be deleted are offered or not. The main check should be whether the modules are in the module planner

* Fix checkstyle errors

* Disallow the setting of UNKNOWN major or focus area

* Add logging to own command parsers

* Fix checkstyle errors

* Add some assertions

* Fix checkstyle

* FocusArea.java: Change continue to return false

Co-Authored-By: GabrielYik <26066169+GabrielYik@users.noreply.github.com>

* Major.java: Change continue to return false

Co-Authored-By: GabrielYik <26066169+GabrielYik@users.noreply.github.com>

* Add year and semester to GoToCommand logging

* DeveloperGuide.adoc: Edit part on manual testing

* Create a new folder for pptx that some images are from

* DeveloperGuide.adoc: Remove mention of Computer Engineering students

* Update colour scheme for commons

* DeveloperGuide.adoc: Add glossary term time-period

* DeveloperGuide.adoc: Add new glossary terms

* DeveloperGuide.adoc: Add new line

* Remove checks for UNKNOWN

GabrielYik committed:
Clear multipurposepanel when clear command is used (#213)

sushinoya commented:
<img width="421" alt="screenshot 2018-11-11 at 10 34 54 pm" src="https://user-images.githubusercontent.com/23443586/48314264-161b2180-e602-11e8-83fd-186508711bc8.png">

I just want to confirm is Computer Engineering is supported now?

sushinoya commented:
<img width="1218" alt="screenshot 2018-11-11 at 10 42 58 pm" src="https://user-images.githubusercontent.com/23443586/48314338-2a135300-e603-11e8-9907-a47c816c3e78.png">

Could have a bit more detail or a screenshot for the status command

sushinoya commented:
<img width="1099" alt="screenshot 2018-11-11 at 10 44 23 pm" src="https://user-images.githubusercontent.com/23443586/48314352-5b8c1e80-e603-11e8-816c-2da283afc29c.png">
Only `add`, `delete` and `clear` are undoable?

sushinoya commented:
Many of the diagrams still say AddressBook in the Developer Guide
<img width="459" alt="screenshot 2018-11-11 at 10 53 53 pm" src="https://user-images.githubusercontent.com/23443586/48314457-c4c06180-e604-11e8-9883-d74cfa9d84a2.png">

rongjiecomputer commented:
> I just want to confirm is Computer Engineering is supported now?

You can input CEG in `setup` command, but `suggest` command will only give generic alpha-sorted list of modules instead of a major-specific suggestion. We deliberately leave it there so that we can demo the difference between major-specific suggestion and generic suggestion.

> Only `add`, `delete` and `clear` are undoable?

Yes.

> Many of the diagrams still say AddressBook in the Developer Guide

Thanks for the catch!

P.S. I thought you will be reviewing our docs yesterday. Maybe I mixed up the date.

Thank you for reviewing our docs at this hour.

sushinoya commented:
You can have syntax highlighting for the codeblocks :)

Hilda-Ang committed:
Update documentations (#212)

* Add project portfolio page

* Rename portfolio and update User Guide

* Update portfolio

* Update Developer Guide

* Update portfolio and User Guide

* Remove duplicated ListSequenceDiagram_1

* Fix inconsistencies in DEveloper Guide and add information on suggestion sorting

* Update DG section in portfolio

* Fix inconsistencies in User Guide and update UG section in portfolio

sushinoya commented:
Final comments:
1. For the Developer Guide, a lot of diagrams are still the old diagrams from AddressBook so you all might wanna update that. Keep those which can be kept and add new ones for things you all have added.
2. Some commands could have been better elaborated on. For example the `add` command's description mostly only enumerates the the different cases of inputs possible.
3. If there's time, could expand more on use cases which there are only three of right now in the Developer Guide. But its probably more important to make sure that the parts above it are well covered first :)

sushinoya commented:
At the end of it all, I am honestly very proud of your group because I think you all attempted something which was not easy and have actually come quite far with it :). Try to fix/polish your app and docs in this last stretch and all the best for your presentation and final exams!

GabrielYik committed:
Update documentation (#218)

* Update personal PPP

* Update personal PPP

* Update documentation

* Remove old ui mock up diagram

* Update sentences and diagram powerpoints for undo redo command

Hilda-Ang committed:
Update documentation (#219)

* Add project portfolio page

* Rename portfolio and update User Guide

* Update portfolio

* Update Developer Guide

* Update portfolio and User Guide

* Remove duplicated ListSequenceDiagram_1

* Fix inconsistencies in DEveloper Guide and add information on suggestion sorting

* Update DG section in portfolio

* Fix inconsistencies in User Guide and update UG section in portfolio

* Change -s for plural forms to s

* Update method names

* Change wording in UG and update portfolio

* Trim portfolio and add syntax highlighting for DG

* Add styling

Hilda-Ang committed:
Miscellaneous (#217)

* Fix comments

* Rename methods

* Add loggers

* Add loggers for parser

* Add logger and rename method

* Remove unused attribute

* Handle invalid cases

* Correct author annotations

* Add helper methods in CommandTestUtil

* Fix style and rename inidces

* Fix indentation levels

* Add comments to isModuleAvailable tests

* Add tests for indexing and remove unused constants

* Format tests

* Remove unused imports

* Add tests for list and suggest command

* Add tests for ModulePlanner

* Fix test in ModulePlanner

* Update comment for SuggestModuleEvent

* Reposition comment in ListModulesEvent

rongjiecomputer committed:
Update UG/DG and use more enums than string (#214)

- Remove unsupported focus areas in UG.
- Add/update docs for setup, storage and moduleInfo.
- `gitignore` Microsoft Office backup files `~$*`.
- Update my PPP (still WIP).
- Refactor `FocusArea` and `Major` for setup command and add a few more tests.

RomaRomama committed:
Update docs and refactor events with 'Module' in it (#216)

* Update DG

* Fix diagram

* Update DG

* Update PPP and UG

* Reformat PPP to only has one major enhancement and update DG

* Fix some sentences

* Fix sequence diagram for status

* Refactor events that has 'Module' in it

* Add forgotten refactoring

* Delete StatusPanelTest

* Added refactoring for ListModulesEvent

* Remove accents

rongjiecomputer committed:
Tweak AsciiDoc style (#220)

GabrielYik committed:
Miscellaneous (#222)

GabrielYik committed:
Documentation (#225)

* DeveloperGuide.adoc: Update images for undo redo portions

* Add activity diagram for iterative deletion

* DeveloperGuide.adoc: Update images

* Remove DeveloperGuide.html

* UserGuide.adoc: Spell out NUS in full for first time use

* Update User Guide

* Update "common" component to be plural

* Update Developer Guide

rongjiecomputer committed:
Update DG and portfolio (#221)

- Tweak wording and fix typos for my portion of DG and portfolio.
- Update size of some images.
- Change block format in `README.adoc`.
- Add manual testing for `setup`.

RomaRomama committed:
Update docs (#224)

* Update add and status sequence diagram

* Add manual testing

* Rename RomaRomama.adoc to romaromama.adoc

* Add s

Hilda-Ang committed:
Update NUS abbreviation

Hilda-Ang committed:
Update documentation (#223)

* Add project portfolio page

* Rename portfolio and update User Guide

* Update portfolio

* Update Developer Guide

* Update portfolio and User Guide

* Remove duplicated ListSequenceDiagram_1

* Fix inconsistencies in DEveloper Guide and add information on suggestion sorting

* Update DG section in portfolio

* Fix inconsistencies in User Guide and update UG section in portfolio

* Change -s for plural forms to s

* Update method names

* Change wording in UG and update portfolio

* Trim portfolio and add syntax highlighting for DG

* Add styling

* Update documentations

* Fix typos and add preface

* Fix formatting in portfolio

* Add project description

* Update images

* Update README to include Computer Engineering

* Add instructions for testing

* Update portfolio

rongjiecomputer committed:
Update README.adoc

GabrielYik committed:
Remove some of my author tags (#227)
